consoleme/handlers/v2/requests.py

```diff
@@ -238,85 +254,104 @@ async def post(self):
             )
             log_data["request"] = extended_request.dict()
             log.debug(log_data)
+
+            if changes.dry_run:
+                response = RequestCreationResponse(
+                    errors=0, request_created=False, extended_request=extended_request
+                )
+                self.write(response.json())
+                await self.finish()
+                return
+
             admin_approved = False
             approval_probe_approved = False
 
-            # TODO: Provide a note to the requester that admin_auto_approve will apply the requested policies only.
-            # It will not automatically apply generated policies. The administrative user will need to visit the policy
-            # Request page to do this manually.
-            if changes.admin_auto_approve:
-                # make sure user is allowed to use admin_auto_approve
-                can_manage_policy_request = (
-                    can_admin_policies(self.user, self.groups),
-                )
-                if can_manage_policy_request:
-                    extended_request.request_status = RequestStatus.approved
-                    admin_approved = True
-                    extended_request.reviewer = self.user
-                    self_approval_comment = CommentModel(
-                        id=str(uuid.uuid4()),
-                        timestamp=int(time.time()),
-                        user_email=self.user,
-                        user=extended_request.requester_info,
-                        last_modified=int(time.time()),
-                        text=f"Self-approved by admin: {self.user}",
-                    )
-                    extended_request.comments.append(self_approval_comment)
-                    log_data["admin_auto_approved"] = True
-                    log_data["request"] = extended_request.dict()
-                    log.debug(log_data)
-                    stats.count(
-                        f"{log_data['function']}.post.admin_auto_approved",
-                        tags={"user": self.user},
-                    )
-                else:
-                    # someone is trying to use admin bypass without being an admin, don't allow request to proceed
-                    stats.count(
-                        f"{log_data['function']}.post.unauthorized_admin_bypass",
-                        tags={"user": self.user},
+            if extended_request.principal.principal_type == "AwsResource":
```

Comment:
This handler method is getting huuuuge. Would love to see it refactored in the future to make it easier to parse.